packages/theme-a11y/theme-a11y.puppeteer.test.js

```diff
@@ -0,0 +1,304 @@
+import {
+  pageLinkFocus,
+  focusable,
+  trapFocus,
+  removeTrapFocus
+} from "./theme-a11y";
+
+describe("trapFocus()", () => {
+  beforeEach(async () => {
+    // This is a super smelly way of exposing the functions needed to perform the
+    // tests I want. Would love a better way!
+    await page.setContent(`
+    <script type="module">
+      window.trapFocusHandlers = {};
+      window.removeTrapFocus = ${removeTrapFocus.toString()};
+      window.pageLinkFocus = ${pageLinkFocus.toString()};
+      window.focusable = ${focusable.toString()};
+      window.trapFocus = ${trapFocus.toString()};
+    </script>
+    <a id="outsideLink" href="#">Some outside link</a>
+    <div id="container">
+      <input id="textInput1" type="text" />
+      <input id="textInput2" type="text" />
+      <button id="button">Some button</button>
+      <button id="hiddenButton" style="display:none;">Hidden button</button>
+    </div>
+    <div id="container2">
+      <input id="textInput3" type="text" />
+    </div>`);
+  });
+
+  test("sets the intial focus on the container by default", async () => {
+    await page.evaluate(() => {
+      const container = document.getElementById("container");
+      window.trapFocus(container);
+    });
+
+    const activeElement = await page.evaluate(() => document.activeElement.id);
+
+    await expect(activeElement).toBe("container");
+  });
+
+  test("accepts an optional second parameter to specify what element to initially focus on", async () => {
+    await page.evaluate(() => {
+      const container = document.getElementById("container");
+      const input2 = document.getElementById("textInput2");
+      window.trapFocus(container, input2);
+    });
+
+    const activeElement = await page.evaluate(() => document.activeElement.id);
+
+    await expect(activeElement).toBe("textInput2");
+  });
+
+  test("focuses the first tabable element after tabbing the last tabable element in a container", async () => {
+    await page.evaluate(() => {
+      const container = document.getElementById("container");
+      window.trapFocus(container);
+    });
+
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+
+    const activeElement = await page.evaluate(() => document.activeElement.id);
+
+    await expect(activeElement).toBe("textInput1");
+  });
+
+  test("focuses on the last tabable and visible element when shift-tabbing from first tabbable element", async () => {
+    await page.evaluate(() => {
+      const container = document.getElementById("container");
+      window.trapFocus(container);
+    });
+
+    await page.keyboard.down("Shift");
```

Comment:
Same thing here.

- focus the first focusable element
- Press shift + tab once